src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -57,19 +59,19 @@ const SearchPage = () => {
         </h1>
       );
     }
-    if (!results) {
-      return (
+    if (!results || searchError) {
```

Comment:
I don't think these are the same thing.  If there's an error, it's worth retrying.  If there are no results, searching again won't produce results.  Let's split these up.

Reply:
While I agree that should be the case, we're talking about a 500 server error no? If an error occurs on the server-side, the user is not obligated to retry because it's something that's out of their control. That's what this issue is specifically targeting, which is why I believe they should be together. I can assign searchError to be the error object that can be used later on though.